test/pages/IndDoYouHaveNINumberPageSpec.scala

```diff
@@ -33,6 +36,17 @@ class IndDoYouHaveNINumberPageSpec extends PageBehaviours {
     booleanField  <- arbitrary[Boolean]
   } yield (addressLookup, address, postcode, name, booleanField)
 
+  private val addressResponseGen: Gen[AddressResponse] = for {
+    addressline  <- arbitrary[String]
+    addressline2 <- arbitrary[Option[String]]
+    addressline3 <- arbitrary[Option[String]]
+    addressline4 <- arbitrary[Option[String]]
+    postcode     <- Gen.option(alphaNumStr.suchThat(_.length >= 5))
```

Comment:
FWIW - in this case `suchThat` should be fine, but I'd generally favour using a sized generator over this approach. That way you generate something you know will be big/small enough each time. Using `suchThat` means something random will be generated, then if it doesn't match the condition it's thrown away and another random value is generated. Sometimes that can be an issue but in this case it should be fine.

Reply:
Good point, in the refactor for the about point ive found a nicer way to generate the postcode parameter, tell me what you think :)